src/Test/L0/Worker/AgentPluginManagerL0.cs

```diff
@@ -0,0 +1,109 @@
+using System;
+using Microsoft.VisualStudio.Services.Agent.Worker;
+using System.Runtime.CompilerServices;
+using Xunit;
+using Pipelines = Microsoft.TeamFoundation.DistributedTask.Pipelines;
+using Agent.Plugins.PipelineArtifact;
+using Agent.Plugins.PipelineCache;
+using System.Collections.Generic;
+
+
+namespace Microsoft.VisualStudio.Services.Agent.Tests.Worker
+{
+    public sealed class AgentPluginManagerL0
+    {
+
+        private class AgentPluginTaskTest
+        {
+            public string Name;
+            public Guid TaskGuid;
+            public List<string> ExpectedTaskPlugins;
+
+            public void RunTest(AgentPluginManager manager)
+            {
+                var taskPlugins = manager.GetTaskPlugins(TaskGuid);
+                Assert.True(taskPlugins.Count == ExpectedTaskPlugins.Count, $"{Name} has {ExpectedTaskPlugins.Count} Task Plugin(s)");
+                foreach (var s in ExpectedTaskPlugins)
+                {
+                    Assert.True(taskPlugins.Contains(s), $"{Name} contains '{s}'");
+                }
+            }
+        }
+
+        [Fact]
+        [Trait("Level", "L0")]
+        [Trait("Category", "Worker")]
+        public void SimpleTests()
+        {
+            using (TestHostContext tc = CreateTestContext())
+            {
+                Tracing trace = tc.GetTrace();
+                var agentPluginManager = new AgentPluginManager();
+                agentPluginManager.Initialize(tc);
+
+                List<AgentPluginTaskTest> tests = new List<AgentPluginTaskTest>
+                {
+                    new AgentPluginTaskTest()
+                    {
+                        Name = "Checkout Task",
+                        TaskGuid = Pipelines.PipelineConstants.CheckoutTask.Id,
+                        ExpectedTaskPlugins = new List<string>
+                        {
+                            "Agent.Plugins.Repository.CheckoutTask, Agent.Plugins",
+                            "Agent.Plugins.Repository.CleanupTask, Agent.Plugins",
+                        }
+                    },
+                    new AgentPluginTaskTest()
+                    {
+                        Name = "Download Pipline Artifact Task",
+                        TaskGuid = PipelineArtifactPluginConstants.DownloadPipelineArtifactTaskId,
+                        ExpectedTaskPlugins = new List<string>
+                        {
+                            "Agent.Plugins.PipelineArtifact.DownloadPipelineArtifactTask, Agent.Plugins",
```

Comment:
Do we want this to have the entire list we had before?

Reply:
I wanted it for this refactor. I wanted to make absolutely sure I didn't miss anything. Going forward we can determine how we want to manage this.

Reply:
Does it have the full list? I think we are missing some.

Reply:
I counted again, there were 14 in the main class and now 14 in the test. I think you might be thinking about extensions, which is controlled in a separate class